### ISSUE TITLE
$where and Javascript Server-side Code Injection

### ISSUE THREAD

**Issue by NelsonBrandao** [strongloop/loopback-connector-mongodb#403]
# Description/Steps to reproduce
MongoDB `$where` operator allows the execution of Javascript code for greater flexibility. This can be a security problem since this connector allows the `$where` operator to be defined on every `filter` param exposed by loopback and by doing so exposing the `$where` filter to the public.

This is valid for every model that extends `PersistedModel` since the usual CRUD endpoint are created with the `filter` query param for every `GET` request.

Something like this:
```
GET /api/brands?filter={"where": {"$where": "if(typeof fzzzuvz==='undefined'){var a=new Date();do{var b=new Date();}while(b-a<5000);fzzzuvz=1;}"}}
```
Allows everyone to delay their request by 5 seconds. Other versions can even cause infinite loop (more information on [here](https://media.blackhat.com/bh-us-11/Sullivan/BH_US_11_Sullivan_Server_Side_WP.pdf)

As a workaround we can disable javascript evaluation in the `$where` operator on the database it self ([here](https://medium.com/mongoaudit/how-to-disable-server-side-javascript-on-mongodb-a099fc0b8f75)) but we lose this functionality on the server as well.

A better solution would be that these operators are parsed or not exposed on the endpoints. This way they remain useful on the API side but disabled from the outside

**Comment by BerkeleyTrue** [comment]
Any response from the @strongloop team on this?

**Comment by NelsonBrandao** [comment]
This issue should not be closed due to inactivity since it's a very serious security bug for everyone using this framework and MongoDB

**Comment by kamilml** [comment]
how can I parse this request before loopback take this code ?

in `server.js` using `app.use()` maybe?

**Comment by tushar-briqsdata** [comment]
This issue should not be closed due to inactivity since it's a very serious security bug for everyone using this framework and MongoDB

**Comment by dhmlau** [comment]
`pinned` it so that it won't get closed by stalebot.

**Comment by virkt25** [comment]
Fix merged. Should be released shortly.

**Comment by virkt25** [comment]
Published as 3.6.0
